### PR TITLE
Renaming ColumnInfo to ColumnOptions

### LINKED ISSUE #2554: ColumnInfo as an API parameter
The use of `ColumnInfo` in the public API can be rather jarring. I'd like to clarify when we should use it in the catalog API, and if the parameter name could be made to be more consistent with the API, such as `ColumnOptions` or `Options`.

### PR DESCRIPTION
**Pull request by artidoro**
Fixes #2554.

As discussed in the issue, we decided to rename `ColumnInfo` to `ColumnOptions`.
I used Visual Studio to find and replace instances of `ColumnInfo`. I did an operation of find and replace for each different casing. I also went over the changes to double check.

### CODE REVIEW

File: src/Microsoft.ML.Data/DataDebuggerPreview.cs [hunk 1]
```diff
@@ -94,14 +94,14 @@ internal RowInfo(int n)
             }
         }
 
-        public sealed class ColumnInfo
+        public sealed class ColumnOptions
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Data/DataDebuggerPreview.cs]: >ColumnOptions [](start = 28, length = 13)

Why?

Comment (sfilipi) [reply src/Microsoft.ML.Data/DataDebuggerPreview.cs]: +1 :)

---
In reply to: [259962854](https://github.com/dotnet/machinelearning/pull/2709#discussion_r259962854) [](ancestors = 259962854)

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs [hunk 2]
```diff
@@ -9,24 +9,24 @@
 
 namespace Microsoft.ML
 {
-    public sealed class SimpleColumnInfo
+    public sealed class ColumnOptions
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: >ColumnOptions [](start = 24, length = 13)

SimpleColumnOptions

File: test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamplesDynamicApi.cs [hunk 3]
```diff
@@ -243,9 +243,9 @@ private void NormalizationWorkout(string dataPath)
             // Apply all kinds of standard ML.NET normalization to the raw features.
             var pipeline =
                 mlContext.Transforms.Normalize(
-                    new NormalizingEstimator.MinMaxColumn("MinMaxNormalized", "Features", fixZero: true),
-                    new NormalizingEstimator.MeanVarColumn("MeanVarNormalized", "Features", fixZero: true),
-                    new NormalizingEstimator.BinningColumn("BinNormalized", "Features", numBins: 256));
+                    new NormalizingEstimator.MinMaxColumnOptions("MinMaxNormalized", "Features", fixZero: true),
```
Comment (artidoro) [inline test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamplesDynamicApi.cs]: Update the cookbook

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs (hunk 2 above)
Comment (artidoro) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: It was intentional, why do you think that `SimpleColumnOptions` is better?
I thought that for consistency it was better like this.

File: src/Microsoft.ML.Data/DataDebuggerPreview.cs (hunk 1 above)
Comment (artidoro) [reply src/Microsoft.ML.Data/DataDebuggerPreview.cs]: Will revert that.

**Pull request merged**